instance/tests/test_ansible.py

```diff
@@ -96,41 +98,105 @@ class AnsibleTestCase(TestCase):
+                requirements_path='/tmp/requirements.txt',
+                playbook_name='playbook_name',
+                remote_username='root',
+                venv_path='/tmp/tempdir/venv'
             )
-            self.assertEqual(
-                mock_popen.mock_calls,
-                [call(run_playbook_cmd, bufsize=1, stdout=-1, stderr=-1, cwd='/play/book', shell=True)]
+
+            mock_popen.assert_called_once_with(
+                "ANSIBLE CMD", bufsize=1, stdout=-1, stderr=-1, cwd='/play/book', shell=True, env=mock.ANY
             )
+
+    def test_render_command(self):
+        """
+        Run the render_sandbox_creation_command function
+        """
+
+        run_playbook_command = ansible.render_sandbox_creation_command(
+            requirements_path='/requirements/path.txt',
+            inventory_path="/tmp/inventory/path",
+            vars_path="/tmp/vars/path",
+            playbook_name='playbook_name',
+            remote_username="root",
+            venv_path='/tmp/venv'
+        )
+        expected = (
+            'virtualenv -p /usr/bin/python /tmp/venv && '
+            '/tmp/venv/bin/python -u /tmp/venv/bin/pip install -r /requirements/path.txt && '
+            '/tmp/venv/bin/python -u /tmp/venv/bin/ansible-playbook -i /tmp/inventory/path '
+            '-e @/tmp/vars/path -u root playbook_name'
+        )
+
+        self.assertEqual(expected, run_playbook_command)
+
+    def test_create_temp_dir_ok(self):
+        """
+        Check if create_temp_dir behaves correctly when no exception is
+        raised from inside it.
+
+        By behave correctly we mean: creates and returns a tempdir, which
+        is deleted after we exit the context manager.
+        """
+        with ansible.create_temp_dir() as temp_dir:
+            self.assertTrue(os.path.exists(temp_dir))
+            self.assertTrue(os.path.isdir(temp_dir))
+        self.assertFalse(os.path.exists(temp_dir))
+
+    def test_create_temp_dir_exception(self):
+        """
+        Check if create_temp_dir behaves correctly when an exception is
+        raised from inside it.
+
+        By behave correctly we mean: creates and returns a tempdir, which
+        is deleted after we exit the context manager.
+        """
+        saved_temp_dir = None
+        try:
+            with ansible.create_temp_dir() as temp_dir:
+                saved_temp_dir = temp_dir
+                self.assertTrue(os.path.exists(temp_dir))
+                self.assertTrue(os.path.isdir(temp_dir))
+                raise KeyboardInterrupt()
+        except KeyboardInterrupt:
+            self.assertFalse(os.path.exists(saved_temp_dir))
+
+    def test_string_to_file_path_ok(self):
+        """
+        Check if string_to_file_path behaves correctly when no exception is
+        raised from inside it.
+
+        By behave correctly we mean: creates a file with proper contents,
+        and this file is deleted when we exit the context manager.
```

Comment:
Right now ``string_to_file_path`` doesn't delete files, there is a previous commit where it did, but then I had `with`` with four objects created. 

Right now a temporary dir is created, and files created by ``test_string_to_file_path_ok`` are created inside it. That directory will be deleted at the end of run.